### PR TITLE
Fail template generation if node version does not match ours

### LINKED ISSUE #1569: create-yoshi-app: A case when user's node version is different than the template's node version
**EDIT: Issue specified here** - https://github.com/wix/yoshi/issues/1569#issuecomment-534976909

# ~~🙋 Feature Request~~
Node 12 is nearing LTS, I think CYA templates should start using node 12 as soon as possible.
A small problem might be that we currently don't switch node versions before testing the templates, so the fact that our templates use the same version of node as `yoshi` does - makes sure we _properly_ test our templates, but if we diverge from `yoshi`'s node version - this'll no longer be the case.
I thought about adding `nvm install && nvm use` before `npm install`ing, and since we only run template e2e's (properly) on CI - we can make the assumption that `nvm` exists.

### 😯 Current Behavior
They use `10`.

### 🤔 Expected Behavior
They'd use `12`.

### PR DESCRIPTION
**Pull request by amitdahan**
### 🔦 Summary
Closes #1569.

Instead of validating for `>= 8.7.0` as we currently do, which allows template generation for node `12` for example, whereas the template to be generated would eventually require `10`, we now fail generation if current node version does not MATCH required version.

This means up until now `create-yoshi-app` supported `>= 8.7.0` and now it (arguably?) correctly checks for `10`.

### 🗺️ Test Plan
- Unit tests for valid and invalid node version.
- Unit tests to make sure all the following match:
`yoshi/.nvmrc`
`yoshi/packages/create-yoshi-app/.nvmrc`
`yoshi/packages/create-yoshi-app/templates/*/javascript/.nvmrc`
`yoshi/packages/create-yoshi-app/templates/*/typescript/.nvmrc`

### CODE REVIEW

**Comment by amitdahan** [comment]
Red until https://github.com/wix/yoshi/pull/1617.

File: packages/create-yoshi-app/src/verifyNodeVersion.ts [hunk 1]
```diff
@@ -0,0 +1,27 @@
+import path from 'path';
+import fs from 'fs';
+import semver from 'semver';
+import chalk from 'chalk';
+
+export default () => {
+  const requiredVersion = fs
+    .readFileSync(
+      path.join(
+        path.dirname(require.resolve('create-yoshi-app/package.json')),
```
Comment (ranyitz) [inline packages/create-yoshi-app/src/verifyNodeVersion.ts]:
```js
const nvmrcPath = path.join(__dirname, '../.nvmrc');
```

File: packages/create-yoshi-app/src/verifyNodeVersion.ts [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+        '.nvmrc',
+      ),
+    )
+    .toString()
+    .trim();
+
+  if (!semver.satisfies(process.version, requiredVersion)) {
+    console.log(
+      `Node version ${chalk.cyan(
+        process.version,
```
Comment (ranyitz) [inline packages/create-yoshi-app/src/verifyNodeVersion.ts]: put into a variable

File: packages/create-yoshi-app/__tests__/verifyNodeVersion.test.ts [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+  });
+};
+
+const readFile = (filePath: string) =>
+  fs
+    .readFileSync(filePath)
+    .toString()
+    .trim();
+
+const nvmrc = readFile(path.resolve(__dirname, '../.nvmrc'));
```
Comment (ranyitz) [inline packages/create-yoshi-app/__tests__/verifyNodeVersion.test.ts]: readNvmrc 😆

**Pull request merged**